forc-plugins/forc-doc/src/render.rs

```diff
@@ -393,9 +397,9 @@ impl Renderable for ItemBody {
 
         let decl_ty = ty_decl.doc_name();
         let friendly_name = ty_decl.friendly_type_name();
-        let sidebar = sidebar.render()?;
-        let item_context = (item_context.context.is_some())
-            .then(|| -> Result<Box<dyn RenderBox>> { item_context.render() });
+        let sidebar = sidebar.render(type_engine.clone())?;
```

Comment:
+1 here

Reply:
Unfortunately, the `box_html!` macro requires literally everything to outlive `'static` 😞

Reply:
Good catch @bingcicle!

The `TypeEngine` can indeed get *massive* with decently sized projects. To give you an idea, recently there was a case where it was only cloned once per package by forc-test, and this turned out to be half of the total cost in terms of time for the total build! It also caused the `sway-applications` (or maybe it was `sway-libraries`?) tests to use ~60G RAM :scream: 

If we really need it to have a `'static` lifetime, we should move it into an `Arc` (or an `Rc` if the `TypeEngine` is not `Sync`) before we start `clone`ing it. This way we only clone the reference counted pointer, not the entire type engine.

Reply:
IIRC it is wrapped in Arc initially

Reply:
Ah right I missed out that part where we're taking in `Arc<TypeEngine>`, oops!